reinforcement_learning/rlclientlib/factory_resolver.cc

```diff
@@ -1,50 +1,61 @@
   namespace u = utility;
-  // For proper static intialization 
+  // For proper static intialization
   // Check https://en.wikibooks.org/wiki/More_C++_Idioms/Nifty_Counter for explanation
   static int init_guard;  // guranteed to be zero when loaded
 
-  // properly aligned memory for the factory object 
+  // properly aligned memory for the factory object
   template <typename T>
   using natural_align = std::aligned_storage<sizeof(T), alignof ( T )>;
 
-  static natural_align<dtfactory>::type dtfactory_buf;
-  static natural_align<modelfactory>::type modelfactory_buf;
+  static natural_align<data_transport_factory_t>::type dtfactory_buf;
+  static natural_align<model_factory_t>::type modelfactory_buf;
+  static natural_align<logger_factory_t>::type loggerfactory_buf;
 
-  // reference should point to the allocated memory to be initalized by placement new in factory_initializer::factory_initializer()
-  dtfactory& data_transport_factory = (dtfactory&)( dtfactory_buf );
-  modelfactory& model_factory = (modelfactory&)( modelfactory_buf );
+  // Reference should point to the allocated memory to be initalized by placement new in factory_initializer::factory_initializer()
+  data_transport_factory_t& data_transport_factory = (data_transport_factory_t&)( dtfactory_buf );
+  model_factory_t& model_factory = (model_factory_t&)( modelfactory_buf );
+  logger_factory_t& logger_factory = (logger_factory_t&)( loggerfactory_buf );
 
   factory_initializer::factory_initializer() {
     if ( init_guard++ == 0 ) {
-      new ( &data_transport_factory ) dtfactory();
-      new ( &model_factory ) modelfactory();
+      new ( &data_transport_factory ) data_transport_factory_t();
+      new ( &model_factory ) model_factory_t();
+      new ( &logger_factory ) logger_factory_t();
 
       register_default_factories();
     }
   }
 
   factory_initializer::~factory_initializer() {
     if ( --init_guard == 0 ) {
-      ( &data_transport_factory )->~dtfactory();
-      ( &model_factory )->~modelfactory();
+      ( &data_transport_factory )->~data_transport_factory_t();
+      ( &model_factory )->~model_factory_t();
+      ( &logger_factory )->~logger_factory_t();
     }
   }
 
   int restapi_data_tranport_create(m::i_data_transport** retval, const u::configuration& cfg, api_status* status);
   int vw_model_create(m::i_model** retval, const u::configuration&, api_status* status);
+  int observation_logger_create(i_logger** retval, const u::configuration&, error_callback_fn*, api_status* status);
+  int interaction_logger_create(i_logger** retval, const u::configuration&, error_callback_fn*, api_status* status);
 
   void factory_initializer::register_default_factories() {
-    data_transport_factory.register_type(value::AZURE_STORAGE_BLOB, restapi_data_tranport_create);
+    data_transport_factory.register_type(value::AZURE_STORAGE_BLOB, restapi_data_tranport_create);data_transport_factory.register_type(value::AZURE_STORAGE_BLOB, restapi_data_tranport_create);
```

Comment:
missing newline?

Reply:
Wow yeah, good catch thanks

---
In reply to: [212092890](https://github.com/JohnLangford/vowpal_wabbit/pull/1575#discussion_r212092890) [](ancestors = 212092890)